### PR TITLE
Start to switch dobi to next-ft-api-client; enable elastic search searching for /somethingId/{base64} endpoints

### PR DESCRIPTION
**Pull request by matthew-andrews**

### CODE REVIEW

File: server/controllers/stream.js [hunk 1]
```diff
@@ -9,46 +9,22 @@ var api = require('next-ft-api-client');
...
-		return article.id;
-	});
+	// HACK:MA: Remove articles that have failed to have been retrieved
+	var ids = posts.filter(function(article) { return !!article; });
 
 	return ft[getItems](ids)
 		.then(function(articles) {
 			var finalizedArticles = [];
-			// HACK:MA: Remove articles that have failed to have been retrieved
-			articles = articles.filter(function(article) {
```
Comment (matthew-andrews) [inline server/controllers/stream.js]: this stuff should have been behind a flag I think…… going to let it “expire” and see the effect……

File: server/controllers/stream.js [hunk 2]
```diff
@@ -9,46 +9,22 @@ var api = require('next-ft-api-client');
...
-				});
-			}
-
 			articles.forEach(function(article, index) {
 				if (article) {
 					if (options.isSearch) {
-						article.matchedString = new MatchHighlighter(options.q, methodeSearchArticles.filter(function(art) {
-							return art.id === article.id;
-						})[0].raw.item.summary.excerpt).run();
+						article.matchedString = new MatchHighlighter(options.q, article.raw.item.summary.excerpt).run();
```
Comment (matthew-andrews) [inline server/controllers/stream.js]: @wheresrhys this _seems_ to achieve the same thing but much more simply……??

File: Makefile [hunk 3]
```diff
@@ -14,7 +15,7 @@ test:
 	nbt verify
 	# Run all tests except for smoke tests
 	$(MAKE) build-production
-	export apikey=12345; export api2key=67890; export ELASTIC_SEARCH_URL='http://123.foundcluster.com:9200/v1_api/item'; export HOSTEDGRAPHITE_APIKEY=123; export ENVIRONMENT=production; mocha --reporter spec tests/server/
+	export apikey=12345; export api2key=67890; export ELASTIC_SEARCH_HOST='ft-elastic-search.com'; export ELASTIC_SEARCH_URL='https://ft-elastic-search.com/v1_api/item'; export HOSTEDGRAPHITE_APIKEY=123; export ENVIRONMENT=production; mocha --reporter spec tests/server/
```
Comment (ironsidevsquincy) [inline Makefile]: do we still use `ELASTIC_SEARCH_URL`?

Comment (matthew-andrews) [inline Makefile]: i think the old
 `ft-api-client` still uses it

**Pull request merged**